exchange/bidder.go

```diff
@@ -501,7 +507,20 @@ func (bidder *bidderAdapter) doRequest(ctx context.Context, req *adapters.Reques
 }
 
 func (bidder *bidderAdapter) doRequestImpl(ctx context.Context, req *adapters.RequestData, logger util.LogMsg) *httpCallInfo {
-	httpReq, err := http.NewRequest(req.Method, req.Uri, bytes.NewBuffer(req.Body))
+	var requestBody []byte
+
+	switch bidder.config.EndpointCompression {
+	case "GZIP":
+		var b bytes.Buffer
+		w := gzip.NewWriter(&b)
+		w.Write([]byte(req.Body))
+		w.Close()
+		requestBody = b.Bytes()
```

Comment:
Lines 514-518 look like a good candidates to be extracted to a separate function. 
`requestBody = compressToGZIP(req.Body)` (or better name for it :))